integration-tests/docker/test_env/cl_node.go

```diff
@@ -0,0 +1,265 @@
+package test_env
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/google/uuid"
+	"github.com/rs/zerolog/log"
+	"github.com/smartcontractkit/chainlink-testing-framework/logwatch"
+	"github.com/smartcontractkit/chainlink/integration-tests/client"
+	env "github.com/smartcontractkit/chainlink/integration-tests/types/envcommon"
+	"github.com/smartcontractkit/chainlink/integration-tests/types/node"
+	"github.com/smartcontractkit/chainlink/integration-tests/utils"
+	"github.com/smartcontractkit/chainlink/integration-tests/utils/templates"
+	tc "github.com/testcontainers/testcontainers-go"
+	tcwait "github.com/testcontainers/testcontainers-go/wait"
+	"math/big"
+)
+
+type ClNode struct {
+	env.EnvComponent
+	API            *client.ChainlinkClient
+	NodeConfigOpts node.NodeConfigOpts
+	DbC            *tc.Container
+	DbCName        string
+	DbOpts         env.PgOpts
+}
+
+func NewClNode(networks []string, opts node.NodeConfigOpts, reusable bool) *ClNode {
+	id := uuid.NewString()
+	co := env.EnvComponentOpts{
+		Name:         "cl-node",
+		ID:           id,
+		ReplicaIndex: opts.ReplicaIndex,
```

Comment:
@skudasov let's have only `ContainerName` in `EnvComponentOpts` as I did in mercury-pipeline repo. This way this is general and not tied to any specification.

```type EnvComponent struct {
	ContainerName string
	Container     tc.Container
	Networks      []string
}

type EnvComponentOpts struct {
	ReuseContainerName string
	Networks           []string
}

func NewEnvComponent(componentType string, opts EnvComponentOpts) EnvComponent {
	var name string
	if opts.ReuseContainerName != "" {
		name = opts.ReuseContainerName
	} else {
		id, _ := uuid.NewRandom()
		name = fmt.Sprintf("%s-%s", componentType, id.String())
	}

	return EnvComponent{
		ContainerName: name,
		Networks:      opts.Networks,
	}
}
```

Reply:
I can move `Reuse` flag out but ID and ReplicaIndex are pretty general too. We may have multiple Geth nodes, CL nodes, etc, wdyt? @lukaszcl

Reply:
@skudasov you should only have full `ContainerName` in the general `EnvComponent` struct. This is the name used to initialize/reuse docker containers. And then you can have construct functions like I have in Mercury 0.3 repo to make sure the new component has a unique name upon initialization:

```
func NewEnvComponent(componentType string, opts EnvComponentOpts) EnvComponent {
	var name string
	if opts.ReuseContainerName != "" {
		name = opts.ReuseContainerName
	} else {
		id, _ := uuid.NewRandom()
		name = fmt.Sprintf("%s-%s", componentType, id.String())
	}

	return EnvComponent{
		ContainerName: name,
		Networks:      opts.Networks,
	}
}
```